modules/ROOT/pages/migration-from-5x.adoc

```diff
@@ -3,88 +3,609 @@
+
+[[plugins-paste]]
+=== Paste
+
+As noted in the xref:6.0-release-notes.adoc#removed-or-deprecated-plugins-paste[_{productname} 6.0 Release Notes_], the `paste` plugin’s functionality is no longer provided as a plugin. It is now a core part of {productname} 6.0.
+
+In conjunction with this change, three changes have been made:
+
+. A `paste` default has changed. The `paste_data_images` option now defaults to `true`.
++
+When `paste` was a plugin, this option was, by default, set to `false`, which prevented images being pasted from the local machine.
++
+. The `mceInsertClipboardContent` argument, `content`, has been renamed. It is now `html`.
++
+The new name is a more accurate reflection of what sort of data the argument passes.
++
+The `content` argument can no longer be used with `mceInsertClipboardContent`. If `content` is used, no data is passed. `mceInsertClipboardContent` only attempts to pass data if the `html` or `text` property is used.
++
+NOTE: `PastePreProcess` also takes an argument, `content`. This argument, for this command, has not been changed.
++
+. The, now built-in, `paste` functionality no longer supports pasting from Microsoft Word.
++
+There is an open https://github.com/tinymce/tinymce/discussions/7487[request for maintainers] should someone in the community be interested in taking over maintainance of this particular functionality as a separate plugin.
++
+If a community-maintained version of the `paste` plugin becomes available, we will link to it from here.
+
+NOTE: The Premium plugin, https://tiny.cloud/tinymce/features/powerpaste[`PowerPaste`], is available. It provides the capability to accept data from Microsoft Word and Microsoft Excel, and clean-up the received data before pasting it into place.
+
+NOTE: No Premium Plugins released with {productname} 6.0 are backwards compatible with earlier versions of {productname}.
+
+== `tabfocus`
+
+The `tabfocus` plugin has been removed. It is no longer part of {productname} 6.0. And it is not otherwise available.
+
+The `tabfocus_elements` option, which relied on the `tabfocus` plugin, no longer functions.
+
+As per standard web practice, the `tabindex` attribute should be used instead of the `tabfocus` plugin or any of the `tabfocus` plugin’s options.
+
+The tab order of a page’s elements, including {productname}, is configured by setting the https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex[`tabindex` attribute] on the relevant HTML elements. The browser then natively handles the tab order.
+
+To configure `tabindex` for the {productname} editor, set the attribute on the target element for the editor. For example:
+
+ ```html
+ tinymce.init({
+   selector: 'textarea'
+ });
+ ...
+ <input tabindex="1">
+ <textarea tabindex="2"></textarea>
+ <input tabindex="3">
+ ```
+In iframe (classic editor) mode, {productname} copies the `tabindex` attribute from the target element to the editor’s iframe, to allow this to work correctly. 
```

Comment:
These highlighted lines are rendering as a single code block. I'm not sure why. Perhaps there should be a newline before the text continues?


![Screen Shot 2022-04-06 at 5 04 12 pm](https://user-images.githubusercontent.com/298292/161915279-4722b060-02d3-4ebf-a0c5-6e51a723c3f9.png)

Reply:
Mitch found this too - it's using markdown syntax, not asciidoc.

Reply:
Code block now marked up as per AsciiDoc, not Markdown.